### PR TITLE
docs: update running locally guide

### PR DESCRIPTION
**Pull request by yiwei-C**
- Update documentation to run TKE locally.
- Add cmd/README.md to describe TKE components.

### CODE REVIEW

File: docs/devel/running-locally.md [hunk 1]
```diff
@@ -1,449 +1,819 @@
...
+  <details>
+  <summary>Click to view sample config</summary>
+  <br>
+
+  ***registry-config.yaml***
+
+  ```yaml
+  apiVersion: registry.config.tkestack.io/v1
+  kind: RegistryConfiguration
+  storage:
```
Comment (yiwei-C) [inline docs/devel/running-locally.md]: @choujimmy What should be the default config for this section?

File: docs/devel/running-locally.md [hunk 2]
```diff
@@ -1,449 +1,819 @@
...
+  - [Go](#go)
+  - [Node.js and NPM](#node-js-and-npm)
+- [Building TKE Components](#building-tke-components)
+- [Create Self-signed Certificates](#create-self-signed-certificates)
+- [Create Static Token](#create-static-token)
+- [Bootstrap TKE Core Components](#bootstrap-tke-core-components)
+  - [tke-auth](#tke-auth)
+  - [tke-platform-api](#tke-platform-api)
+  - [tke-platform-controller](#tke-platform-controller)
+  - [tke-registry-api(Optional)](#tke-registry-api-optional-)
```
Comment (John-Lin) [inline docs/devel/running-locally.md]: tke-registry-api seems not an optional component at installation process. tke-installer will push images to registry through registry-api after tke components are deployed.

Comment (yiwei-C) [reply docs/devel/running-locally.md]: Thanks for the response. 

They are different processes to launch TKE. This "running locally" guide covers the process of running TKE core components one by one on local machine for development purpose while the `tke-installer` gives you an easy way to install TKE stack and includes some of our best practice. We made the "running locally" process more flexible and you can choose whether you want to enable `tke-registry-api`.

**Comment by John-Lin** [comment]
Put `README.md` under `cmd` folder would cause build binaries error.
Screen Shot 2019-12-19 at 9 01 53 PM

**Comment by yiwei-C** [comment]
> Put `README.md` under `cmd` folder would cause build binaries error. how about move to `docs`?
> 
> Screen Shot 2019-12-19 at 9 01 53 PM

Fixed by adding a filter. Let me know if you think the doc should not be placed in the `cmd` folder.

**Comment by John-Lin** [comment]
LGTM

**Pull request merged**